mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -818,45 +818,61 @@ def __repr__(self):
 
 
 @PIPELINES.register_module()
-class IndoorPointSample(object):
-    """Indoor point sample.
+class PointSample(object):
+    """Point sample.
 
     Sampling data to a certain number.
 
     Args:
-        name (str): Name of the dataset.
         num_points (int): Number of points to be sampled.
+        sample_range (int, optional): The range where to sample points.
     """
 
-    def __init__(self, num_points):
+    def __init__(self, num_points, sample_range=None):
         self.num_points = num_points
-
-    def points_random_sampling(self,
-                               points,
-                               num_samples,
-                               replace=None,
-                               return_choices=False):
+        self.sample_range = sample_range
+
+    def _points_random_sampling(self,
+                                points,
+                                num_samples,
+                                sample_range=None,
+                                replace=None,
+                                return_choices=False):
         """Points random sampling.
 
         Sample points to a certain number.
 
         Args:
             points (np.ndarray | :obj:`BasePoints`): 3D Points.
             num_samples (int): Number of samples to be sampled.
-            replace (bool): Whether the sample is with or without replacement.
-            Defaults to None.
-            return_choices (bool): Whether return choice. Defaults to False.
-
+            sample_range (int, optional): Indicating the range where the points
+                will be sampled.
+                Defaults to None.
+            replace (bool, optional): Sampling with or without replacement.
+                Defaults to None.
+            return_choices (bool, optional): Whether return choice.
+                Defaults to False.
         Returns:
             tuple[np.ndarray] | np.ndarray:
-
                 - points (np.ndarray | :obj:`BasePoints`): 3D Points.
                 - choices (np.ndarray, optional): The generated random samples.
         """
         if replace is None:
             replace = (points.shape[0] < num_samples)
-        choices = np.random.choice(
-            points.shape[0], num_samples, replace=replace)
+        point_range = range(len(points))
+        if sample_range:
+            depth = points.coord[:, 2]
+            far_inds = np.where(depth > sample_range)[0]
+            near_inds = np.where(depth <= sample_range)[0]
+            if not replace:
+                # Only sampling the near points when len(points) >= num_samples
+                point_range = near_inds
+                num_samples -= len(far_inds)
+        choices = np.random.choice(point_range, num_samples, replace=replace)
+        if sample_range and not replace:
```

Comment:
Would it be better if we use `if sample_range is not None`? After all it's possible value is `int` or `None`. I understand that `if None` is equal to `if False` but I think `not None` is a better style?

Reply:
Sorry for the late reply. I accidentally missed this review before.

If we sample points from point clouds containing few points, then the sample range will not take effect. And all points, including far points and near points, need to be sampled with the option `replace`.

Reply:
Sorry for the late reply. I accidentally missed this review before.

If we sample points from point clouds containing few points, then the sample range will not take effect. And all points, including far points and near points, need to be sampled with the option replace.

Reply:
Oh, I mean changing it to `if sample_range  is not None and not replace:` . My concern is that `sample_range` is not a `bool` variable, so directly using `if` on it is not standard?